clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -77,6 +84,13 @@
     private static final String SUBSCRIPTION_EXCEPTION_MESSAGE =
         "Subscription to topics, partitions and pattern are mutually exclusive";
 
+    private void confirmSubscriptionType(SubscriptionType type) {
```

Comment:
I was having some doubts when picking this name. The method both sets the state (if not set yet) and verifies it (if set). I'll change it to `setSubscriptionType`. Thanks.